biglog/segment.go

```diff
@@ -164,6 +173,9 @@ func loadSegment(indexPath string) (seg *segment, err error) {
 	seg.setNextOffsets()
 	seg.setCreatedTS()
 
+	// TODO: when this is called via createSegment the index is
+	// created by createSegIndex which doesn't write this timestamp?!
```

Comment:
the timestamp is added at data write time. The way the index works is that the last entry is always the next one (without timestamp), this means the next offset and the next pointer in the data is pre-defined in the index. It’s not the most intuitive flow but it ended up being like that for convenience after a few iterations.